src/ucp/core/ucp_am.c

```diff
@@ -282,13 +276,14 @@ ucs_status_t ucp_worker_set_am_recv_handler(ucp_worker_h worker,
 static UCS_F_ALWAYS_INLINE ssize_t
 ucp_am_bcopy_pack_data(void *buffer, ucp_request_t *req, size_t length)
 {
-    size_t total_length      = 0ul;
-    ucp_dt_state_t hdr_state = {};
+    size_t total_length = 0ul;
```

Comment:
maybe avoid total_length var? seems like we could add req->send.msg_proto.am.header_length to the result of ucp_dt_pack() of the data. Maybe just add local var for "user_header_length" since its used several times